File: src/main/kotlin/org/geepawhill/contentment/core/GroupSource.java
package org.geepawhill.contentment.core;

import javafx.scene.Group;

public interface GroupSource 
{
	Group group();

	GroupSource NONE = () -> {
		throw new NoGroupSource();
	};
	
	class NoGroupSource extends RuntimeException
	{
		public NoGroupSource()
		{
			super("GroupSource has no group.");
		}
	}

}

File: src/main/java/org/geepawhill/contentment/core/GroupSource.kt
package org.geepawhill.contentment.core

import javafx.scene.Group

interface GroupSource {
    fun group(): Group

    class NoGroupSource : RuntimeException("GroupSource has no group.")

    companion object {
        val NONE: GroupSource = object : GroupSource {
            override fun group(): Group {
                throw NoGroupSource()
            }
        }
    }

}
